src/main/java/com/mkopec/ticketbookingapp/service/SeatValidationService.java

```diff
@@ -0,0 +1,199 @@
+            switch (p.seatType) {
+                case FREE_INSIDE: {
+                    if (hasPrevious(modelIndex - 1)) {
+                        SeatModel pp = getPrevious(neighbourhood, modelIndex - 1);
+                        switch (pp.seatType) {
+                            case FREE_INSIDE:
+                            case FREE_ON_EDGE:
+                                isFreeSeatLeft = false;
+                                break;
+                            case OCCUPIED:
+                            case RESERVATION_DONE:
+                                isFreeSeatLeft = true;
+                                break;
+                        }
+                    }
+                    break;
+                }
+                case FREE_ON_EDGE:
+                    isFreeSeatLeft = true;
+                    break;
+                case OCCUPIED:
+                case RESERVATION_DONE:
+                    return true;
+            }
+        }
+
+        if (isFreeSeatLeft) {
+            if (hasNext(neighbourhood, modelIndex)) {
+                SeatModel n = getNext(neighbourhood, modelIndex);
+                switch (n.seatType) {
+                    case FREE_INSIDE:
+                    case FREE_ON_EDGE:
+                        return false;
+                    case OCCUPIED:
+                    case RESERVATION_DONE:
+                        break;
+                }
+            }
+        } else {
+            if (hasNext(neighbourhood, modelIndex)) {
+                SeatModel n = getNext(neighbourhood, modelIndex);
+                switch (n.seatType) {
+                    case FREE_INSIDE: {
+                        if (hasNext(neighbourhood, modelIndex + 1)) {
+                            SeatModel nn = getNext(neighbourhood, modelIndex + 1);
+                            switch (nn.seatType) {
+                                case FREE_INSIDE:
+                                case FREE_ON_EDGE:
+                                    break;
+                                case OCCUPIED:
+                                case RESERVATION_DONE:
+                                    return false;
+                            }
+                        }
+                        break;
+                    }
+                    case FREE_ON_EDGE:
+                    case OCCUPIED:
+                    case RESERVATION_DONE:
+                        break;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    private SeatModel getNext(List<SeatModel> neighbourhood, int index) {
+        return neighbourhood.get(index + 1);
+    }
+
+    private SeatModel getPrevious(List<SeatModel> neighbourhood, int index) {
+        return neighbourhood.get(index - 1);
+    }
+
+    private boolean hasPrevious(int index) {
+        return index - 1 >= 0;
+    }
+
+    private boolean hasNext(List<SeatModel> neighbourhood, int index) {
+        return index + 1 < neighbourhood.size();
+    }
```

Comment:
te 4 metody wyglądają jakby zawsze dotyczyły tego samego. Może można by je zamknąć w jakiejś jednej klasie? Jest coś takiego jak programowanie obiektowe.

Reply:
Racja, mogłem to opakować, ewentualnie zamienić na abstrakcję z typem generycznym